### PR TITLE
ADH-4393 - [UI] Pagination

### PR DESCRIPTION
**Pull request by dbardin90**
https://tracker.yandex.ru/ADH-4393

### CODE REVIEW

File: smart-frontend/app/src/components/uikit/Pagination/Pagination.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+const PaginationExample = ({ totalItems, pageData, perPageItems, isNextBtn, hidePerPage }: PaginationProps) => {
+  const [curPageData, setCurPageData] = useState(pageData);
+
+  const handleOnChange = ({ perPage, pageNumber }: PaginationParams) => {
+    setCurPageData({ perPage, pageNumber });
+  };
+
+  return (
+    <Pagination
+      onChangeData={handleOnChange}
```
Comment (KuzminIA) [inline smart-frontend/app/src/components/uikit/Pagination/Pagination.stories.tsx:70]: please, rename to `handleChange`

File: smart-frontend/app/src/components/uikit/Pagination/Pagination.tsx [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+  };
+
+  useEffect(() => {
+    if (totalItems > 0 && totalItems <= pageNumber * perPage) {
+      setPageNumber(Math.ceil(totalItems / perPage) - 1);
+    }
+  }, [totalItems, pageNumber, perPage, setPageNumber]);
+
+  const handleResetFiltersClick = () => {
+    console.info('reset');
```
Comment (KuzminIA) [inline smart-frontend/app/src/components/uikit/Pagination/Pagination.tsx:109]: please, remove

File: smart-frontend/app/src/components/uikit/Pagination/Pagination.tsx [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+          {frequencyComponent}
+        </>
+      )}
+      <div className={s.pagination__buttonWrapper} data-test="pagination-button-container">
+        <RenderNumberButtons setPageNumber={setPageNumber} items={pageItems} currentPageNumber={pageNumber} />
+        {totalPages === 0 && (
+          <PaginationStepButton arrowVariant={'arrowDouble'} onClick={() => setPageNumber(0)} disabled={!hasPrev} />
+        )}
+        <PaginationStepButton
+          arrowVariant={'arrowSingle'}
```
Comment (KuzminIA) [inline smart-frontend/app/src/components/uikit/Pagination/Pagination.tsx:141]: please, change to ```arrowVariant="arrowSingle"```

File: smart-frontend/app/src/components/uikit/Pagination/Pagination.tsx [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+        />
+        <PaginationStepButton
+          arrowVariant={'arrowSingle'}
+          onClick={() => setPageNumber(pageNumber + 1)}
+          variant={'next'}
+          disabled={!hasNext}
+          dataTest="pagination-next-page"
+        />
+        <PaginationStepButton
+          arrowVariant={'arrowDouble'}
```
Comment (KuzminIA) [inline smart-frontend/app/src/components/uikit/Pagination/Pagination.tsx:154]: please, change to ```arrowVariant="arrowSingle"```

File: smart-frontend/app/src/components/uikit/Pagination/Pagination.tsx [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+        <PaginationStepButton
+          arrowVariant={'arrowSingle'}
+          onClick={() => setPageNumber(pageNumber - 1)}
+          disabled={!hasPrev}
+          dataTest="pagination-prev-page"
+        />
+        <PaginationStepButton
+          arrowVariant={'arrowSingle'}
+          onClick={() => setPageNumber(pageNumber + 1)}
+          variant={'next'}
```
Comment (KuzminIA) [inline smart-frontend/app/src/components/uikit/Pagination/Pagination.tsx:149]: please, change to ```variant="next"```

File: smart-frontend/app/src/components/uikit/Pagination/Pagination.stories.tsx (hunk 1 above)
Comment (dbardin90) [reply smart-frontend/app/src/components/uikit/Pagination/Pagination.stories.tsx:70]: fixed

File: smart-frontend/app/src/components/uikit/Pagination/Pagination.tsx (hunk 2 above)
Comment (dbardin90) [reply smart-frontend/app/src/components/uikit/Pagination/Pagination.tsx:109]: fixed

File: smart-frontend/app/src/components/uikit/Pagination/Pagination.tsx (hunk 3 above)
Comment (dbardin90) [reply smart-frontend/app/src/components/uikit/Pagination/Pagination.tsx:141]: fixed

File: smart-frontend/app/src/components/uikit/Pagination/Pagination.tsx (hunk 4 above)
Comment (dbardin90) [reply smart-frontend/app/src/components/uikit/Pagination/Pagination.tsx:154]: fixed

File: smart-frontend/app/src/components/uikit/Pagination/Pagination.tsx (hunk 5 above)
Comment (dbardin90) [reply smart-frontend/app/src/components/uikit/Pagination/Pagination.tsx:149]: fixed

**Review by MrHalless — APPROVED** [review]

**Comment by tigrulya-exe** [comment]
Please, rebase on the master and run the [following command](https://github.com/arenadata/SSM/blob/develop/docs/ssm-developer-guide.md#license-header-update) to add license headers to the new source files. Otherwise, subsequent CI builds will fail.

**Review by rimakan — APPROVED** [review]

**Review by KuzminIA — APPROVED** [review]

**Review by remizov-arena — APPROVED** [review]

**Pull request merged**